src/problem-solver.cc

```diff
@@ -328,6 +339,34 @@ namespace hpp {
       }
     }
 
+    void ProblemSolver::addConfigValidationBuilder
+    (const std::string& type, const ConfigValidationBuilder_t& builder)
+    {
+      configValidations.add (type, builder);
+    }
```

Comment:
A comment to explain the rationale behind public members of type `Container`: I made them public in order to avoid the need of duplicating the containers API for each of them.